pkg/controller/reconciler/util.go

```diff
@@ -280,3 +280,19 @@ func createOrUpdateContainer(ctx context.Context, radius RadiusClient, container
 
 	return nil, nil
 }
+
+func generateDeploymentResourceName(resourceId string) string {
+	resourceBaseName := strings.Split(resourceId, "/")[len(strings.Split(resourceId, "/"))-1]
```

Comment:
Can we use the resource ID parser for this? It's hard to reason about where this data came from, and whether we expect it to be valid.

For example if I gave you the string `""` this would cause a panic. Any manual string parsing code like this needs a lot of testing, so we should avoid it.

Reply:
changed this to use the resourceid parser